numba/listobject.py

```diff
@@ -86,13 +86,23 @@ class ListStatus(IntEnum):
     LIST_ERR_ITER_EXHAUSTED = -4
 
 
-def _raise_if_error(context, builder, status, msg):
-    """Raise an internal error depending on the value of *status*
+class ErrorHandler(object):
+    """ErrorHandler for calling codegen functions from this file.
+
+    Stores the state needed to raise an exception from nopython mode.
     """
-    ok_status = status.type(int(ListStatus.LIST_OK))
-    with builder.if_then(builder.icmp_signed('!=', status, ok_status),
-                         likely=True):
-        context.call_conv.return_user_exc(builder, RuntimeError, (msg,))
+
+    def __init__(self, context):
+        self.context = context
+
+    def __call__(self, builder, status, msg):
+        ok_status = status.type(int(ListStatus.LIST_OK))
+        with builder.if_then(builder.icmp_signed('!=', status, ok_status),
+                             likely=True):
+            self.context.call_conv.return_user_exc(
+                builder, RuntimeError, (msg,))
+
+
```

Comment:
flake8 complains about too many blank lines here now.

Reply:
yeah, not sure how that made it through my `pre-commit` hooks.

Reply:
fixed and pushed